nix/haskell.nix

```diff
@@ -54,7 +54,7 @@ let
         packages.binary.configureFlags = [ "--ghc-option=-Werror" ];
         #packages.binary/test.configureFlags = [ "--ghc-option=-Werror" ];
         packages.cardano-crypto-class.configureFlags = [ "--ghc-option=-Werror" ];
-        packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [[ buildPackages.libsodium-vrf ]];
+        packages.cardano-crypto-class.components.library.pkgconfig = lib.mkForce [[ buildPackages.libsodium-vrf buildPackages.secp256k1 ]];
```

Comment:
`Ptr Void` corresponds to `void*` in C, which is 'pointer to _anything_'. This isn't in any way only inhabitable by `NULL`, but by _literally anything ever_.

Regarding initialization costs: the `secp256k1` library itself claims contexts should be re-used, rather than re-created, due to performance costs. This means it's probably not _negligible_, but _how_ not-negligible is hard to assess.

`capi` is good for two reasons:

* It ensures we obey the platform's calling convention; and
* It checks that we don't accidentally stuff mismatched types into a C function.

The second part of this functionality [is a bit broken on our GHC](https://gitlab.haskell.org/ghc/ghc/-/issues/15531), and I'm unsure if the `SizedPtr` tricks we do help in any way, as they wrap `void*` which can be cast to absolutely anything. The main reason @lehins brought up `ccall unsafe` was probably performance: I don't know if `capi` is costlier.